### PR TITLE
Addresses #4717 Add fb authentication using oauth

### LINKED ISSUE #4717: Add third party authentication using facebook oauth
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-server

**Current behavior:**

Currently, there isn't any Server API to connect to facebook using oauth as required at `open-event-frontend`

**Expected behavior:**

Addition of connect to open event with facebook. 

**Steps to reproduce:**

**Related code:**

```
insert any relevant code here else remove this section
```

**Other information:**

**System information:** 

```
Your operating system
```

```
output of `python --version`
```
**Wanna work on this issue**

Parent issue: #5115

### PR DESCRIPTION
**Pull request by bhaveshAn**
Fixes #4717 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:

Added fb oauth, receiving access token.

#### Changes proposed in this pull request:

- Adds endpoint `/oauth/<provider>` to return redirect URI for FE.
- Adds endpoint `/oauth/token/<provider>` to authenticate and return the Access token along with token type i.e. Bearer.
-

fb

@srv-twry @mayank8318 @schedutron @maxlorenz @Kreijstal 
FB authentication is complete to best of my knowledge.
The response shows access_token and token_type, though the access_token is having black marks to hide it a bit due to security issues. So please don't mind. :smile: 
Please review. 
Also, this is a priority issue and a big feature too. So it would be good to split it and take responsibilities individually. In that case this PR is ready to merge too. Please see the issue as well. #4717

### CODE REVIEW

**Comment by srv-twry** [comment]
@bhaveshAn Please update this as early as you can so that I can do the FE integration for it. I would need some time for it too hence please update this quickly.

**Comment by mariobehling** [comment]
@bhaveshAn Please follow up on the requests and comments here and finalize this PR. Please ask others to review it.

File: app/api/auth.py [hunk 1]
```diff
@@ -14,10 +14,26 @@
 from app.models.notification import PASSWORD_CHANGE as PASSWORD_CHANGE_NOTIF
 from app.models.user import User
 from app.api.helpers.errors import UnprocessableEntityError, NotFoundError, BadRequestError
+from app.api.helpers.third_party_auth import GoogleOAuth, FbOAuth, TwitterOAuth, InstagramOAuth
 
 auth_routes = Blueprint('auth', __name__, url_prefix='/v1/auth')
 
 
+@auth_routes.route('/oauth/<provider>', methods=['GET'])
```
[the remark this answers is not in the document]
Comment (srv-twry) [reply app/api/auth.py]: Correct according to me.

**Comment by srv-twry** [comment]
> could do the blueprint as well theoretically

@Kreijstal This is a key feature and is required. @bhaveshAn can add it in another PR. I'll merge as soon as the build passes.

**Pull request merged**